crates/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -814,6 +814,47 @@ pub(super) fn concat_hor(
     delimiter: &str,
     ignore_nulls: bool,
 ) -> PolarsResult<Series> {
+    // If delimiter is empty and ignore_nulls is false, we'll check for nulls in the data
+    if delimiter.is_empty() && !ignore_nulls {
```

Comment:
This is redundant as ignore_nulls cant be true unless delimiter is empty
You should choose one location to do this check, I vote here because it ensures this condition if called from anywhere other than concat_str

Reply:
removed `&& !ignore_nulls`